packages/inventory/src/containers/InventoryAddEdit/index.tsx

```diff
@@ -0,0 +1,267 @@
+import { InventoryItemForm, defaultInitialValues } from '../../components/InventoryItemForm';
+
+/** register reducers */
+reducerRegistry.register(locationUnitsReducerName, locationUnitsReducer);
+reducerRegistry.register(productCatalogueReducerName, ProductCatalogueReducer);
+reducerRegistry.register(inventoryReducerName, inventoryReducer);
+
+/** Route params */
+interface RouteParams {
+  [ROUTE_PARAM_SERVICE_POINT_ID]: string;
+  [ROUTE_PARAM_INVENTORY_ITEM_ID]?: string;
+}
+
+/** interface component props */
+export interface InventoryAddEditProps extends RouteComponentProps<RouteParams> {
+  products: ProductCatalogue[];
+  openSRPBaseURL: string;
+  cancelURL: string;
+  redirectURL: string;
+  fetchProductsCreator: typeof fetchProducts;
+  fetchLocationUnitsCreator: typeof fetchLocationUnits;
+  fetchInventoriesCreator: typeof fetchInventories;
+  servicePoint: LocationUnit | null;
+  inventory?: Inventory | null;
+  customFetchOptions?: typeof getFetchOptions;
+}
+
+/** default component props */
+export const defaultInventoryAddEditProps = {
+  openSRPBaseURL: '',
+  cancelURL: '',
+  redirectURL: '',
+  products: [],
+  fetchProductsCreator: fetchProducts,
+  fetchLocationUnitsCreator: fetchLocationUnits,
+  fetchInventoriesCreator: fetchInventories,
+  servicePoint: null,
+};
+
+const InventoryAddEdit: React.FC<InventoryAddEditProps> = (props: InventoryAddEditProps) => {
+  const {
+    openSRPBaseURL,
+    cancelURL,
+    products,
+    fetchProductsCreator,
+    fetchLocationUnitsCreator,
+    fetchInventoriesCreator,
+    redirectURL,
+    servicePoint,
+    inventory,
+    match,
+    customFetchOptions,
+  } = props;
+  const [UNICEFSections, setUNICEFSections] = React.useState<Setting[]>([]);
+  const [donors, setDonors] = React.useState<Setting[]>([]);
+  const { Title } = Typography;
+  const isEdit = !!match.params[ROUTE_PARAM_INVENTORY_ITEM_ID];
+
+  useEffect(() => {
+    // Handle when servicePoint is null e.g when a user refreshes page
+    if (!servicePoint) {
+      // fetch location units
+      const service = new OpenSRPService(
+        LOCATIONS_GET_ALL_SYNC_ENDPOINT,
+        openSRPBaseURL,
+        customFetchOptions
+      );
+      service
+        .list({
+          serverVersion: 0,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          is_jurisdiction: false,
+        })
+        .then((response: LocationUnit[]) => {
+          fetchLocationUnitsCreator(response);
+        })
+        .catch((_: HTTPError) => {
+          sendErrorNotification(ERROR_GENERIC);
+        });
+    }
+  }, [servicePoint, fetchLocationUnitsCreator, openSRPBaseURL, customFetchOptions]);
+
+  useEffect(() => {
+    if (!products.length) {
+      loadProductCatalogue(
+        openSRPBaseURL,
+        OpenSRPService,
+        fetchProductsCreator
+      ).catch((_: HTTPError) => sendErrorNotification(ERROR_GENERIC));
+    }
+  }, [products.length, openSRPBaseURL, fetchProductsCreator]);
```

Comment:
does this hold the chance that we could deal with stale data here, i.e. we will not fetch data if there are some in the store.

Reply:
@p-netm Ok...but it also means the user must have fetched the products at some point within the application, should we also handle that scenario for the other pieces of code? Or what would be the purpose of the store in such cases?

Reply:
@kelvin-muchiri could you say that in another way

Reply:
@p-netm Given that the store does resides in memory so we can't really argue that it will be available for long such as a cookie...If there are products in the store, then they must have been fetched within the period the user visited the site and it's ok to fetch from store

Reply:
@p-netm Otherwise, when do we use the store and when do we not?

Reply:
yeah, but we cannot guarantee that during that time from when the user initially fetches the product-catalogue, to when they try to access this page, that the data in the api will not have changed by other means.

Reply:
@p-netm Noted...I will change to make a fetch

Reply:
> Otherwise, when do we use the store and when do we not?

@p-netm

Reply:
we are still using the store to drive all UI changes, we should however try to make sure that we have the most updated information as we do so.